include/ap/argument_parser.hpp

```diff
@@ -36,6 +36,7 @@ class argument_interface {
     virtual const value_type& value() const = 0;
 
     virtual const std::string_view name() const = 0;
+    virtual const std::optional<std::string_view> short_name() const = 0;
```

Comment:
This should not be in the interface, as the short name is unique to the optional argument.
Ultimately we want to have an `argument_name` structure and an abstract function returning this name object, but now we return a `string_view` due to the lack of such class.
Still this functions returns an identifier of an argument, so the `short_name` functions is reduntant